### ISSUE TITLE
[question] Publish updates to Nova Extension directory

### ISSUE THREAD

**Issue by andreipetre** [guillaumealgis/Serpens.novaextension#4]
@guillaumealgis Thank you for your work on this extension.

I wanted to ask if the changes from this year have been published to the Nova Extension directory?

I'm asking as the extension still shows version 2.0.0 from 10 months ago, despite there being commits from Aug and Sep 2023.

![Nova Extension](https://github.com/guillaumealgis/Serpens.novaextension/assets/1036164/7f0a0d6c-fc92-41e3-9880-f8de5c2af5b4)

Also, it doesn't look like all configuration fields are available on Nova. From my understanding, [`Basic options`](https://github.com/guillaumealgis/Serpens.novaextension/blob/main/extension.json) should contain a `Python virtualenv path` setting.

Here's the general settings screen for Serpents that I see

<img width="827" alt="Basic Options" src="https://github.com/guillaumealgis/Serpens.novaextension/assets/1036164/1f735f62-52f0-4b46-8551-da1ad291ae15">

and the project configuration which only contains the `Workspace Jedi Env Path` setting.

<img width="604" alt="Project Options" src="https://github.com/guillaumealgis/Serpens.novaextension/assets/1036164/28d5d58f-6aa8-4c4d-906c-f1a1700c235d">

---

Perhaps my understanding of these settings is incorrect? 🙃 

Is there a way to install this extension directly from Github without waiting on the Nova Extension directory to be updated?

Thank you.

**Comment by guillaumealgis** [comment]
Hello @andreipetre  thanks for your interest in Serpens !

> I wanted to ask if the changes from this year have been published to the Nova Extension directory?

They are not, as the work in `main` are still very much a work in progress.

I know the 2.0.0 version of the extension currently published in the Nova Extension directory is not in a great state, it's basically all the same code as the upstream project Serpens was born from : [mmshivesh's Python extension](https://github.com/mmshivesh/Python-Nova.novaextension), which was abandoned.

I'm currently working on a rework of the extension, but have hit a couple of problems with Nova's implementation of the Language Server Protocol that I need to find workarounds for.

If you wish to test things our yourself, you can install the extension from Github by following the [CONTRIBUTING guide](https://github.com/guillaumealgis/Serpens.novaextension/blob/main/CONTRIBUTING.md) :

1. `git clone https://github.com/guillaumealgis/Serpens.novaextension.git`
2. `cd Serpens.novaextension`
3. `brew install npm`
4. `npm install`
5. `npx tsc`
6. Then open the Serpens.novaextension, either by double clicking it in Finder, or with `open /path/to/Serpens.novaextension`

This should install your local version of the extension in Nova (you may need to uninstall the published version of the extension from Nova before installing the local one).

**⚠️ This may or may not work ! As I said, this is very much a work in progress.**

If things don't work out at all : 
- try opening the new Serpens Sidebar (View > Sidebars > Show All Sidebars > then select "Serpens")
- try reading logs in the Extension Console (Extensions > Show Extensions Console)

**Comment by andreipetre** [comment]
Hi @guillaumealgis,

Thanks for the explanation.

I installed the extension from this repo in Nova. It works fine for me with [`python-lsp-server`](https://pypi.org/project/python-lsp-server/) 1.7.4 on Nova 11.6 (build 551099).

When I use `python-lsp-server` 1.8.0 or above, I always get the same error in the extension console:

```
Python Language Server error: The data couldn’t be read because it isn’t in the correct format.
```

Thank you.

**Comment by guillaumealgis** [comment]
@andreipetre yes, it's the same issue as https://github.com/guillaumealgis/Serpens.novaextension/issues/5

I haven't got around to look at it yet, but I'm not sure that's something I can fix on my end anyway :/
